Source/Engine/Physics/Actors/RigidBody.cpp

```diff
@@ -300,6 +300,46 @@ void RigidBody::ClosestPoint(const Vector3& position, Vector3& result) const
     }
 }
 
+void RigidBody::AddMovement(const Vector3& translation, const Quaternion& rotation)
+{
+    // filter rotation according to constraints
+    Quaternion allowedRotation;
+    if (static_cast<unsigned int>(GetConstraints()) & static_cast<unsigned int>(RigidbodyConstraints::LockRotation))
+        allowedRotation = Quaternion::Identity;
+    else
+    {
+        Float3 euler = rotation.GetEuler();
+        const auto constraints = static_cast<unsigned int>(GetConstraints());
+        if (constraints & static_cast<unsigned int>(RigidbodyConstraints::LockRotationX))
```

Comment:
Use `EnumHasAnyFlags(GetConstraints(). RigidbodyConstraints::LockRotationX)` for readability (the same for Y/Z axes and translation axes too).